tests/ui/lint/warn-path-statement.rs

```diff
@@ -14,4 +14,12 @@ fn main() {
 
     let z = (Droppy,);
     z; //~ ERROR path statement drops value
+
+    macro_rules! foo {
+        ($e:expr) => {
+            $e;
+        };
+    }
+
+    foo!(x);
```

Comment:
@jieyouxu is this what you meant? Right now this is emitting the lint. Is the idea that this lint shouldn't be emitted the lint in this case or that it shouldn't be machine applicable?

Reply:
I somehow completely lost this notification, sorry about that. Now that I read my own comment https://github.com/rust-lang/rust/pull/140830#pullrequestreview-2827781740 back, I think what I had in mind at the time was stuff like having a path statement that comes from a macro expansion from another crate, where I think I thought that a machine-applicable suggestion in that case wouldn't be very actionable.

... But that can be a follow-up or further adjusted as necessary, and does not need to block this PR at all.